File: Ticket.Data/Models/Ticket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;
using Ticket.Data.Models.Enums;

namespace Ticket.Data.Models
{
    public class Ticket
    {
        public int TicketId { get; set; }
        public int ProductId { get; set; }
        public string TicketDescription { get; set; }      
        public string Title { get; set; }
        //public string Attachment { get; set; }
        public Status status { get; set; }
        public DateTime? ClosedAt { get; set; } = null;
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ModifiedAt { get; set; } = null;
        public int? ClosedBy { get; set; } = 0;
        public int? AssignedTo { get; set; } = 0;
        public Priorities? Priority { get; set; } = null;

        public int ReminderLevel { get; set; } = 0;
        public int? CreatedBy { get; set; }
        public Product Product { get; set; }
        public User? User { get; set; } 
        public List<Comment> Comments { get; set; } = new List<Comment>();
        public List<Attachment> Attachments { get; set; } = new List<Attachment>();

    }
}

File: ticket-system/src/app/core/model/Ticket.ts
export class Ticket {
  ticketId: number;
  productId: number;
  ticketDescription: string;
  title: string;
  attachment: string[];
  fileName: string[];
  status: number;
  closedAt: string;
  createdAt: string;
  modifiedAt: string;
  closedBy: number;
  assignedTo: number;
  priority: number;
  createdBy: number;
  userId: number;
  userName: string;

  constructor() {
    this.ticketId = 0;
    this.productId = 0;
    this.ticketDescription = '';
    this.title = '';
    this.attachment = [];
    this.status = 0;
    this.closedAt = '0001-01-01T00:00:00';
    this.createdAt = '';
    this.modifiedAt = '';
    this.closedBy = 0;
    this.assignedTo = 0;
    this.priority = 0;
    this.createdBy = 0;
    this.userId = 0;
    this.userName = '';
    this.fileName = [];
  }
}
